src/package/subscriberPackageVersion.ts

```diff
@@ -0,0 +1,503 @@
+   * Installs a package version in a subscriber org.
+   *
+   * Package Version install emits the following events:
+   * - PackageEvents.install.warning
+   * - PackageEvents.install.presend
+   * - PackageEvents.install.postsend
+   * - PackageEvents.install['subscriber-status']
+   *
+   * @param pkgInstallCreateRequest
+   * @param options
+   */
+  public async install(
+    pkgInstallCreateRequest: PackageInstallCreateRequest,
+    options: PackageInstallOptions = allZeroesInstallOptions
+  ): Promise<PackagingSObjects.PackageInstallRequest> {
+    try {
+      // before starting the install, check to see if the package version is available for install
+      await waitForPublish(
+        this.connection,
+        await this.getId(),
+        options.publishFrequency,
+        options.publishTimeout,
+        pkgInstallCreateRequest.Password
+      );
+      const pkgVersionInstallRequest = await createPackageInstallRequest(
+        this.connection,
+        pkgInstallCreateRequest,
+        await this.getPackageType()
+      );
+      return SubscriberPackageVersion.installStatus(
+        this.connection,
+        pkgVersionInstallRequest.Id,
+        pkgInstallCreateRequest.Password,
+        options
+      );
+    } catch (e) {
+      throw applyErrorAction(massageErrorMessage(e as Error));
+    }
+  }
+
+  /**
+   * Uninstalls a package version from a subscriber org.
+   *
+   * @param frequency
+   * @param wait
+   */
+  public async uninstall(
+    frequency: Duration = Duration.milliseconds(0),
+    wait: Duration = Duration.milliseconds(0)
+  ): Promise<PackagingSObjects.SubscriberPackageVersionUninstallRequest> {
+    return await uninstallPackage(await this.getId(), this.connection, frequency, wait);
+  }
+
+  /**
+   * Returns an array of RSS and CSP external sites for the package.
+   *
+   * @param installationKey The installation key (if any) for the subscriber package version.
+   * @returns an array of RSS and CSP site URLs, or undefined if the package doesn't have any.
+   */
+  public async getExternalSites(): Promise<Optional<string[]>> {
+    getLogger().debug(`Checking package: [${await this.getId()}] for external sites`);
+
+    const remoteSiteSettings = await this.getRemoteSiteSettings();
+    const cspTrustedSites = await this.getCspTrustedSites();
+
+    const rssUrls = remoteSiteSettings?.settings ? remoteSiteSettings.settings?.map((rss) => rss.url) : [];
+    const cspUrls = cspTrustedSites?.settings ? cspTrustedSites?.settings.map((csp) => csp.endpointUrl) : [];
+
+    const sites = [...rssUrls, ...cspUrls];
+    return sites.length > 0 ? sites : undefined;
+  }
+
+  private async getFieldValue<T>(field: string): Promise<T> {
+    if (!Reflect.has(this, field)) {
+      await this.getData({ includeHighCostFields: highCostQueryFields.includes(field) });
+    }
+    return Reflect.get(this, field);
+  }
+
+  private getFieldsForQuery(options: { force?: boolean; includeHighCostFields?: boolean }): string[] {
```

Comment:
Using the full list of fields provide of fields that need to be queried.
It resolves if high cost query fields should be included
It retains fields that not been read force === false
It retains fields if force === true